Content.Client/Chat/ChatManager.cs

```diff
@@ -104,5 +128,84 @@ private void _onChatBoxTextSubmitted(ChatBox chatBox, string text)
                 }
             }
         }
+
+        private void _onFilterButtonToggled(ChatBox chatBox, Button.ButtonToggledEventArgs e)
+        {
+            // TODO make toggled ALL button flip all button states programatically + visually
+            switch (e.Button.Name)
+            {
+                case "Local":
+                    _Localstate = !_Localstate;
+                    if (_Localstate)
+                    {
+                        _filteredChannels = ChatChannel.Local;
+                        RepopulateChat(filteredHistory);
+                        break;
+                    }
+                    else
+                    {
+                        _filteredChannels = _filteredChannels ^ ChatChannel.Local;
```

Comment:
You want to use `&= ~` here. `^=` toggles the flag. I assume you're trying to flat unset it here.

```suggestion
                        _filteredChannels &= ~ChatChannel.Local;
```